src/plugins/vis_type_table/public/types.ts

```diff
@@ -38,22 +39,46 @@ export enum AggTypes {
   COUNT = 'count',
 }
 
-export interface Dimensions {
-  buckets: SchemaConfig[];
+export interface TableVisConfig extends TableVisParams {
+  title: string;
   metrics: SchemaConfig[];
+  buckets: SchemaConfig[];
+  splitRow?: SchemaConfig[];
+  splitColumn?: SchemaConfig[];
 }
 
 export interface TableVisParams {
-  type: 'table';
   perPage: number | '';
   showPartialRows: boolean;
   showMetricsAtAllLevels: boolean;
-  sort: {
-    columnIndex: number | null;
-    direction: string | null;
-  };
   showTotal: boolean;
   totalFunc: AggTypes;
   percentageCol: string;
-  dimensions: Dimensions;
+}
+
+export interface FormattedColumn {
+  id: string;
+  title: string;
+  formatter: IFieldFormat;
+  filterable: boolean;
+  formattedTotal?: string | number;
+  sumTotal?: number;
+  total?: number;
+}
+
+export interface ColumnWidth {
+  colIndex: number;
+  width: number;
+}
+
+export interface SortColumn {
```

Comment:
Why not `ColumnSort`?

Reply:
hhh no particular reason. just think the purpose of this interface is to sort column

Reply:
Yeah, at least in the context of this file, `ColumnSort` would be nicer with `ColumnWidth`.

Reply:
will change to `ColumnSort`